cmd/soroban-rpc/internal/db/event.go

```diff
@@ -78,15 +100,60 @@ func (eventHandler *eventHandler) InsertEvents(lcm xdr.LedgerCloseMeta) error {
 		}
 
 		query := sq.Insert(eventTableName).
-			Columns("id", "ledger_sequence", "application_order", "contract_id", "event_type")
+			Columns(
+				"id",
+				"contract_id",
+				"event_type",
+				"event_data",
+				"ledger_close_time",
+				"transaction_hash",
+				"topic1",
+				"topic2",
+				"topic3",
+				"topic4",
+			)
 
 		for index, e := range txEvents {
-			var contractId []byte
+
+			var contractID []byte
 			if e.Event.ContractId != nil {
-				contractId = e.Event.ContractId[:]
+				contractID = e.Event.ContractId[:]
 			}
-			id := events.Cursor{Ledger: lcm.LedgerSequence(), Tx: tx.Index, Op: 0, Event: uint32(index)}.String()
-			query = query.Values(id, lcm.LedgerSequence(), tx.Index, contractId, int(e.Event.Type))
+
+			id := Cursor{Ledger: lcm.LedgerSequence(), Tx: tx.Index, Op: 0, Event: uint32(index)}.String()
+			eventBlob, err := xdr.NewEncodingBuffer().MarshalBinary(&e)
+			if err != nil {
+				return err
+			}
+
+			v0, ok := e.Event.Body.GetV0()
+			if !ok {
+				return errors.New("unknown event version")
+			}
+
+			// Encode the topics
+			maxTopicCount := 4
+			topicList := make([]string, maxTopicCount)
+			for index, segment := range v0.Topics {
+				seg, err := xdr.MarshalBase64(segment)
```

Comment:
Why marshal topics? We can compare bytes in the SQL query instead to save space

Reply:
Good call out !! refactored :)